.github/workflows/duckinuous-quackration.yml

```diff
@@ -86,3 +85,76 @@ jobs:
       - name: Run flake8
         run: "flake8 \
         --format='::error file=%(path)s,line=%(row)d,col=%(col)d::[flake8] %(code)s: %(text)s'"
+
+  build:
+    name: Build & Quack
+    runs-on: ubuntu-latest
+
+    needs: [lint]
+    if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+
+    steps:
+      - name: Create SHA Container Tag
+        id: sha_tag
+        run: |
+          tag=$(cut -c 1-7 <<< $GITHUB_SHA)
+          echo "::set-output name=tag::$tag"
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Github Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN  }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          cache-from: type=registry,ref=ghcr.io/python-discord/quackstack:main
+          tags: |
+            ghcr.io/python-discord/quackstack:main
+            ghcr.io/python-discord/quackstack:${{ steps.sha_tag.outputs.tag }}
+
+  deploy:
```

Comment:
We should add something here to limit this to master, not PRs, and possibly use an environment.

Reply:
Nevermind, missed it in the build. Still the point about environments.

Reply:
This has already the `needs: [build]` and `build` job defines constraint, so if build gets skipped, then this skips too.